src/components/common/Dialogs/ControlSafeDialog/SafeTransactionPreview.tsx

```diff
@@ -290,117 +280,126 @@ const SafeTransactionPreview = ({
           colony={colony}
         />
       </DialogSection>
-      {transactions.map((transaction, index) => (
-        <div key={autogeneratedIds[index]}>
-          {transactions.length > 1 && (
-            <div
-              className={classnames(styles.transactionHeading, {
-                [styles.transactionHeadingOpen]: transactionTabStatus[index],
-              })}
-            >
-              <Heading
-                appearance={{
-                  size: 'normal',
-                  margin: 'none',
-                  theme: 'dark',
-                }}
-                text={MSG.transactionTitle}
-                textValues={{
-                  transactionNumber: index + 1,
-                  transactionType: getTransactionTypeLabel(
-                    transaction.transactionType,
-                  ),
-                }}
-              />
-              <Button
-                className={styles.tabButton}
-                onClick={() => handleTabToggle(index)}
+      {transactions.map((transaction, index) => {
+        if (transaction.transactionType === undefined) {
+          throw new Error(
+            `transaction type of: ${transaction} is undefined... something has gone terribly wrong`,
+          );
+        }
```

Comment:
Ok, I see what you mean 👍 

I think in this case it's better to change the type to `transactionType: SafeTransactionType | undefined` because it's not that the `transactionType` field is optional (It's always gonna be present in any of the transactions.), but that it can be `undefined` if the user hasn't reached that point in the form.

What do you think?